lib/zipkin/samplers/probabilistic.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module Zipkin
+  module Samplers
+    # Probabilistic sampler
+    #
+    # Sample a portion of traces using trace_id as the random decision
+    class Probabilistic
+      def initialize(rate: 0.001)
+        if rate < 0.0 || rate > 1.0
+          raise "Sampling rate must be between 0.0 and 1.0, got #{rate.inspect}"
+        end
+        @boundary = TraceId::TRACE_ID_UPPER_BOUND * rate
+      end
+
+      def sample?(trace_id)
+        @boundary >= trace_id.to_i(16)
```

Comment:
So trace IDs are already random hex numbers? Does this decision need to be made multiple times for the same trace? I.e. is it important that the decision is deterministic, given a certain trace?

Reply:
Yes, they are basically `rand(2**64)`. There's no need to do it multiple times. It's done once and then propagated using http headers or whatever the transport is. So I don't think it has to be deterministic but I don't think it hurts. I followed https://github.com/jaegertracing/jaeger-client-go/blob/master/sampler.go#L110